### PR TITLE
[Node] support openai SDK version ^4.24.1 embeddings api

### PR DESCRIPTION
**Pull request by koolamusic**
**Changes:**
- Updated `OpenAIEmbeddingFunction` to align with OpenAI SDK v4.24.
- Removed the configuration parameter from the OpenAI client initialization for simplicity.
- Implemented Sinon stubs in tests to mock `openai.embeddings.create` method, eliminating the need for a real API key.

**Impact:**
- Streamlines OpenAI client usage in line with the latest SDK.

### CODE REVIEW

**Comment by changhiskhan** [comment]
hey there are you able to just run `npm run lint -- --fix` to fix the lint issues?

File: node/package.json
Comment (changhiskhan) [inline node/package.json:49]: could go to 4.24.3 now

**Review by changhiskhan — COMMENTED** [review]
looks good in general. Thanks so much!
Should be good to go once you resolve the merge conflicts.

excited to get this update in.

File: node/package.json
Comment (koolamusic) [reply node/package.json:49]: @changhiskhan done

File: node/src/embedding/openai.ts
Comment (changhiskhan) [inline node/src/embedding/openai.ts:42]:
```suggestion
```

Remove this console log

**Review by changhiskhan — APPROVED** [review]
looks good other than the console log. thanks!

**Comment by changhiskhan** [comment]
ok i fixed the merge conflicts. and merged this manually. thanks for the contrib

**Comment by koolamusic** [comment]
My pleasure.

**Pull request merged**
